runners/core-java/src/test/java/org/apache/beam/runners/core/SplittableParDoTest.java

```diff
@@ -146,18 +151,23 @@ public void testBoundednessForUnboundedFn() {
         "Applying an unbounded SDF to a bounded collection produces a bounded collection",
         PCollection.IsBounded.UNBOUNDED,
         makeBoundedCollection(pipeline)
-            .apply("unbounded to bounded", new SplittableParDo<>(unboundedFn))
+            .apply("unbounded to bounded", new SplittableParDo.Bound<>(ParDo.of(unboundedFn)))
             .isBounded());
     assertEquals(
         "Applying an unbounded SDF to an unbounded collection produces an unbounded collection",
         PCollection.IsBounded.UNBOUNDED,
         makeUnboundedCollection(pipeline)
-            .apply("unbounded to unbounded", new SplittableParDo<>(unboundedFn))
+            .apply("unbounded to unbounded", new SplittableParDo.Bound<>(ParDo.of(unboundedFn)))
```

Comment:
This is quite an interesting situation. `DoFnTester` is for users, and it plays the role of the runner/worker in this regard.

So for a user who actually wants to test their splittable `DoFn`, the conversion to use any of the technology you have included should occur within `DoFnTester`, privately. But, of course, `DoFnTester` cannot access `runners-core`. I am not familiar enough with the preceding work on splittable `DoFn` to know how this should be resolved.

Often (as you can see from the internals in `DoFnTester`) we end up needing a slight variant on the details of what would otherwise come from `runners-core`. So in this case having a near-duplicate implementation there is not so bad. In the case of a splittable `DoFn` I would expect that the `DoFnTester` code to exercise splittability would diverge even further from this particular implementation in terms of `StateInternals`, etc.

If we stick to hacking on `setXYZ` methods, I would prefer to add `DoFnTester.getOutputWindowedValue(...)` accessor and `ProcessFn.setOutputWindowedValue(...)`, or appropriate variations. This will limit the use of this pattern to just testing, while all worker-side uses will be able to pass things through the constructor without a mutator.